jose/jwt.py

```diff
@@ -53,7 +86,16 @@ def encode(claims, key, algorithm=ALGORITHMS.HS256, headers=None, access_token=N
     return jws.sign(claims, key, headers=headers, algorithm=algorithm)
 
 
-def decode(token, key, algorithms=None, options=None, audience=None, issuer=None, subject=None, access_token=None):
+def decode(
+    token: str,
+    key: Union[str, Dict[str, Any]],
```

Comment:
The comment at https://github.com/mpdavis/python-jose/blob/master/jose/jwt.py#L61 says that key can be an individual jwk or set of jwks. This doesn't allow for a set to be passed. Also, although it isn't in the comment, a list also works since `_get_keys()` checks if the key is iterable at https://github.com/mpdavis/python-jose/blob/be8e914a63e8940cd34ac0f5a066d114f10dad48/jose/jws.py#L242 . 

IMO, this should accept a string, dictionary, or anything iterable to be backwards compatible.